local-setup/jupyter/crs_loader.py

```
@@ -0,0 +1,1363 @@
+    def __init__(self, base_url: str):
+        """Initialize CRS Loader with DIGIT environment URL
+
+        Args:
+            base_url: DIGIT gateway URL (e.g., "https://unified-dev.digit.org")
+        """
+        self.base_url = base_url.rstrip('/')
+        self.uploader: Optional[APIUploader] = None
+        self.tenant_id: Optional[str] = None
+        self._authenticated = False
+
+    def login(self, username: str = None, password: str = None,
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)